### ISSUE TITLE
Test t/race-conditions.t seems to last infinitely on PPC Mac OS X 10.5.8, Leopard

### ISSUE THREAD

**Issue by ballapete** [miyagawa/CGI-Compile#27]
Testing happens in the MacPorts package manager, launched in GNU Emacs as root, t.psgi.t has finished, and while t/race-conditions.t seems not to be able to find its end I send a HUP signal to it:

```
--->  Testing p5.34-cgi-compile
Executing:  cd "/opt/local/var/macports/build/_Volumes_Halde_ports_perll_p5-cgi-compile/p5.34-cgi-compile/work/CGI-Compile-0.26" && /opt/local/bin/perl5.34 Build test 
t/00_compile.t ......... ok
t/alarm.t .............. ok
t/author-pod-syntax.t .. skipped: these tests are for testing by the author
t/chained_exit.t ....... ok
t/chdir.t .............. ok
t/coderef_args.t ....... ok
t/compile.t ............ ok
t/data_end.t ........... ok
t/eval-my-variables.t .. ok
t/exit-code.t .......... ok
t/exit.t ............... ok
t/local-SIG.t .......... ok
t/psgi.t ............... ok
t/race-conditions.t .... 
Failed 40/75 subtests 
t/return-val.t ......... ok
t/source.t ............. ok
t/source_filter.t ...... ok
t/symbols.t ............ ok
t/warnings.t ........... ok

Test Summary Report
-------------------
t/race-conditions.t  (Wstat: 1 Tests: 35 Failed: 0)
  Non-zero wait status: 1
  Parse errors: Bad plan.  You planned 75 tests but ran 35.
Files=19, Tests=171, 268 wallclock secs ( 0.50 usr  0.32 sys + 16.52 cusr  4.88 csys = 22.22 CPU)
Result: FAIL
```

pstree shows:

```
root 290 /\ pstree -w -p 32778
-+= 00001 root /sbin/launchd
 \-+- 00365 pete /bin/sh /opt/local/bin/startx
   \-+- 00420 pete xinit /Users/pete/.xinitrc -- /opt/local/bin/X :0 -listen tcp -iglx -dpi 96 -auth /Users/pete/.serverauth.365
     \-+= 00578 pete /opt/local/bin/blackbox
       \-+- 00588 pete /usr/X11/bin/xterm -geometry 80x50+625+125 -fa courier -fs 7 -tn xterm-256color
         \-+= 00594 pete -tcsh
           \-+= 32778 macports /opt/local/libexec/macports/bin/tclsh8.6 /opt/local/bin/port -vs test p5.34-cgi-compile
             \-+- 32785 macports sh -c  cd "/opt/local/var/macports/build/_Volumes_Halde_ports_perll_p5-cgi-compile/p5.34-cgi-compile/work/CGI-Compile-0.26" && /opt/local/bin/perl5.34 Build test 
               \-+- 32791 macports /opt/local/bin/perl5.34 Build test
                 \--- 32810 macports /opt/local/bin/perl5.34 t/race-conditions.t
root 291 /\ kill -1 32810
```

On more up-to-date Macs with intel CPUs testing takes a few seconds. Lack of computational power should not make the test endure for minutes, half an hour.

**Comment by rkitover** [comment]
@miyagawa I have also noticed problems with this test in CPANTesters reports, how about we make this one author-only?

**Comment by miyagawa** [comment]
sounds reasonable.

**Comment by rkitover** [comment]
All set, the fix is in 0.27 and it's on CPAN.
